packages/ui/src/views/repo/hooks/use-view-management.ts

```diff
@@ -0,0 +1,320 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { FilterValue, SavedView, SortValue, ValidationResult, ViewManagement } from '@components/filters/types'
+
+interface UseViewManagementProps {
+  storageKey: string
+  setActiveFilters: (filters: FilterValue[]) => void
+  setActiveSorts: (sorts: SortValue[]) => void
+}
+
+/**
+ * Compares two arrays for equality in the context of filter and sort states
+ * Used to detect changes in filter/sort configurations and view states
+ *
+ * @param {Array} arr1 - First array (usually current filters/sorts)
+ * @param {Array} arr2 - Second array (usually saved view filters/sorts)
+ * @returns {boolean} Whether the arrays have the same elements regardless of order
+ *
+ * @example
+ * // Used to detect changes in view state
+ * const hasViewChanges = !areArraysEqual(currentView.filters, activeFilters) ||
+ *                       !areArraysEqual(currentView.sorts, activeSorts)
+ */
+export const areArraysEqual = (arr1: any[], arr2: any[]) => {
+  if (!arr1 || !arr2) return false
+  if (arr1.length !== arr2.length) return false
+
+  // Sort arrays to ensure order-independent comparison
+  const sorted1 = [...arr1].sort((a, b) => JSON.stringify(a).localeCompare(JSON.stringify(b)))
+  const sorted2 = [...arr2].sort((a, b) => JSON.stringify(a).localeCompare(JSON.stringify(b)))
+
+  return JSON.stringify(sorted1) === JSON.stringify(sorted2)
+}
+
+/**
+ * Hook for managing filter views state and operations
+ * Handles saving, loading, updating and validating filter views
+ *
+ * @param storageKey - Unique key for localStorage
+ * @param setActiveFilters - Callback to update active filters
+ * @param setActiveSorts - Callback to update active sorts
+ */
+export const useViewManagement = ({
```

Comment:
this hook implements logic that seems specific to harness open-source, and should be implemented in the gitness app instead.

Reply:
this should also be implemented using zustand stores once it's in gitness